helix-view/src/view.rs

```diff
@@ -12,42 +12,48 @@ pub const PADDING: usize = 5;
 
 type Jump = (DocumentId, Selection);
 
+#[derive(Default)]
 pub struct JumpList {
     jumps: Vec<Jump>,
     current: usize,
 }
 
 impl JumpList {
-    pub fn new(initial: Jump) -> Self {
-        Self {
-            jumps: vec![initial],
-            current: 0,
-        }
-    }
-
     pub fn push(&mut self, jump: Jump) {
```

Comment:
I think this should be fixable by keeping the old code, but instead adjust the `undo` to push the current position as a jump.

Kakoune works the same way, pushing the old position before the jump (not after): https://github.com/mawww/kakoune/blob/d758bbf09b949f0fef88ef3c6762cc55098cdb95/src/normal.cc#L235

Pushes the current position before an undo: https://github.com/mawww/kakoune/blob/d758bbf09b949f0fef88ef3c6762cc55098cdb95/src/buffer.cc#L287

Reply:
This is to fix the behavior of having the first jump stuck even though you have jump from first jump. So in this case the first jump will always get stuck in the same position even though you ever use it anymore. Kakoune have this behavior but I find it weird.